rust-counter/src/processor.rs

```diff
@@ -0,0 +1,297 @@
+
+    Ok(())
+}
+
+pub fn process_increase_counter(
+    program_id: &Pubkey,
+    accounts: &[AccountInfo],
+    increase_by: u64
+) -> ProgramResult {
+    // Iterating accounts
+    let accounts_iter = &mut accounts.iter();
+    let initializer_account = next_account_info(accounts_iter)?;
+    let counter_account = next_account_info(accounts_iter)?;
+    let system_program = next_account_info(accounts_iter)?;
+
+    let (counter_pda, bump_seed) = Pubkey::find_program_address(
+        &[b"counter_account", initializer_account.key.as_ref()],
+        program_id,
+    );
+
+    // Check to ensure that you're using the right PDA
+    if counter_pda != *counter_account.key {
+        msg!("Invalid seeds for PDA");
+        return Err(ProgramError::InvalidArgument);
+    }
+
+    // Get and borrow lamports from greeting account
+    let borrowed_lamports = counter_account.try_borrow_lamports().unwrap();
+
+    // The account must be owned by the program in order to modify its data
+    if counter_account.owner != program_id {
+        msg!("Counter account owner: {} does not have the correct program id {}", counter_account.owner, program_id);
+        return Err(ProgramError::IncorrectProgramId);
+    }
+
+    // Increment
+    let mut counter_data = Counter::try_from_slice(&counter_account.data.borrow())?;
+    msg!("Increase count");
+    counter_data.count += increase_by;
+    counter_data.serialize(&mut &mut counter_account.data.borrow_mut()[..])?;
+    msg!(
+        "PDA {} count: {}",
+        counter_account.key,
+        counter_data.count
+    );
+
+    Ok(())
+}
+
+pub fn process_delegate(
+    program_id: &Pubkey,
+    accounts: &[AccountInfo],
+) -> ProgramResult {
+
+    // Get accounts
+    let account_info_iter = &mut accounts.iter();
+    let initializer = next_account_info(account_info_iter)?;
+    let system_program = next_account_info(account_info_iter)?;
+    let pda_to_delegate = next_account_info(account_info_iter)?;
+    let owner_program = next_account_info(account_info_iter)?;
+    let delegation_buffer = next_account_info(account_info_iter)?;
+    let delegation_record = next_account_info(account_info_iter)?;
+    let delegation_metadata = next_account_info(account_info_iter)?;
+    let delegation_program = next_account_info(account_info_iter)?;
+    
+    // Prepare counter pda seeds
+    let seed_1 = b"counter_account";
+    let seed_2 = initializer.key.as_ref();
+    let pda_seeds: &[&[u8]] = &[seed_1, seed_2];
+
+    let delegate_accounts = DelegateAccounts {
+        payer: initializer,
+        pda: pda_to_delegate,
+        owner_program,
+        buffer: delegation_buffer,
+        delegation_record,
+        delegation_metadata,
+        delegation_program,
+        system_program,
+    };
+
+    let delegate_config = DelegateConfig {
+        commit_frequency_ms: 30_000,
+        validator: None,
+    };
+
+    delegate_account(delegate_accounts, &pda_seeds, delegate_config);
```

Comment:
Should propagate the result:

delegate_account(delegate_accounts, &pda_seeds, delegate_config)?;